server/pulp/server/async/tasks.py

```diff
@@ -255,7 +255,7 @@ def apply_async_with_reservation(self, resource_type, resource_id, *args, **kwar
         # To avoid the race condition where __call__ method below is called before
         # this change is propagated to all db nodes, using an 'upsert' here and setting
         # the task state to 'waiting' only on an insert.
-        task_status.save(fields_to_set_on_insert=['state', 'start_time'])
+        task_status.save(fields_to_set_on_insert=['state', 'start_time'], safe=True)
```

Comment:
When I and @skarmark originally wrote all the task system DB code, we used safe=True on all of our pymongo queries. That is what I meant when I said we originally used safe=True. I think it's imperative that we continue doing safe queries as we transition to Mongoengine, as we had such safe queries when we used pymongo directly.

The reason that ```write_concern={fsync: True}``` is not equivalent to the old safe=True is that users can configure default write_concern on their replica sets. I confirmed that this was possible at least as early as mongo 2.2:

http://docs.mongodb.org/v2.2/applications/replication/#modify-default-write-concern

So if a user sets the default write concern to the majority of their nodes, this code will override that default and we will not have a safe write. The safe=True that we used to use respected the default write concern of the replica set.